services/w3c/w3c-validation.service.js

```diff
@@ -0,0 +1,269 @@
+'use strict'
+const url = require('url')
+const Joi = require('@hapi/joi')
+const { optionalUrl } = require('../validators')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  url: optionalUrl.required(),
+  messages: Joi.array()
+    .required()
+    .items(Joi.object()),
+}).required()
```

Comment:
This is the schema that is used to validate the response from the the upstream API (The W3C Checker Web Service in this case) so it should model the response objects that we care about. 

As such, I don't think we need `url` here right? The `messages` are the only things we care about. We should also be more explicit about those message objects, including the properties we expect to see on the objects in the `messages` array

Reply:
Remove url from schema

Reply:
Excellent. What about the schema for the message objects inside `messages`? 

I'm guessing it would result in the schema looking something like:
```js
const schema = Joi.object({
  messages: Joi.array()
    .required()
    .items(Joi.object({
      message: Joi.allow('error', 'warning', '...').required(),
    })),
}).required()
```

Reply:
Added validation for the `type` property